src/main/scala/beam/agentsim/agents/household/HouseholdCAVScheduling.scala

```diff
@@ -1,195 +1,398 @@
+    // adding a time window to the total travel time
+    HouseholdTrips(
+      requests.sortWith(_.time < _.time).toList,
+      totTravelTimeOut + sumTimeWindows,
+      tripTravelTime.toMap,
+      totTravelTimeOut
+    )
+  }
+
+  def getDefaultMode(legOption: Option[Leg], nbVehicles: Int): BeamMode = {
+    legOption
+      .flatMap(leg => BeamMode.fromString(leg.getMode))
+      .getOrElse(if (nbVehicles <= 0) BeamMode.TRANSIT else BeamMode.CAR)
   }
-  requests = requests.sortWith(_.time < _.time)
-  def apply(): List[MobilityServiceRequest] = { requests }
-  override def toString = s"${requests}"
 }
 
 class HouseholdCAVScheduling(
-  plans: List[BeamPlan],
-  fleetSize: Int,
-  timeWindow: Double,
-  skim: Map[Coord, Map[Coord, Double]]
+  val population: org.matsim.api.core.v01.population.Population,
+  val household: Household,
+  val householdVehicles: List[BeamVehicle],
+  val pickupTimeWindow: Double,
+  val dropoffTimeWindow: Double,
+  val skim: Map[BeamMode, Map[Coord, Map[Coord, Double]]]
 ) {
-  private var fleet = List[HouseholdCAV]()
-  private var feasibleSchedules = List[HouseholdSchedule]()
 
-  case class HouseholdCAV(id: Int, maxOccupancy: Int)
+  // ***
+  def getAllFeasibleSchedules(): List[CAVFleetSchedule] = {
+    import beam.agentsim.agents.memberships.Memberships.RankedGroup._
+    implicit val pop: org.matsim.api.core.v01.population.Population = population
+    val householdPlans = household.members.map(person => BeamPlan(person.getSelectedPlan))
+    // extract potential household CAV requests from plans
+    val householdRequests: HouseholdTrips =
+      HouseholdTrips(householdPlans, householdVehicles.size, pickupTimeWindow, dropoffTimeWindow, skim)
 
-  class CAVSchedule(val schedule: List[MobilityServiceRequest], cav: HouseholdCAV, val cost: Double, occupancy: Int) {
-    var feasible: Boolean = true
+    // initialize household schedule
+    val cavVehicles = householdVehicles.filter(_.beamVehicleType.automationLevel > 3)
+    import scala.collection.mutable.{ListBuffer => MList}
+    val emptyFleetSchedule = MList.empty[CAVSchedule]
+    cavVehicles.foldLeft(householdRequests.requests.head) {
+      case (req, cav) =>
+        emptyFleetSchedule.prepend(
+          new CAVSchedule(
+            List[MobilityServiceRequest](
+              new MobilityServiceRequest(
+                None,
+                req.activity,
+                req.time,
+                req.trip,
+                req.defaultMode,
+                Init,
+                req.time
+              )
+            ),
+            cav,
+            0
+          )
+        )
+        (req)
+    }
+    // compute all the feasible schedules through
+    val feasibleSchedules = MList[CAVFleetSchedule](new CAVFleetSchedule(emptyFleetSchedule.toList, householdRequests))
+    for (request  <- householdRequests.requests;
+         schedule <- feasibleSchedules) yield {
```

Comment:
There is no need to yield here because the end result is not used. It will translate into a nested foreach without the yield, which seems to be the intent